content/client-lib-development-guide/features.textile

```diff
@@ -502,7 +502,8 @@ h3(#realtime-channel). Channel
 *** @(RTL6c5)@ A publish should not trigger an implicit attach (in contrast to earlier version of this spec)
 ** @(RTL6d)@ Messages for a single channel that have been queued may be sent in a single @ProtocolMessage@ by bundling them into the @ProtocolMessage#messages@ array, subject to the following constraints:
 *** @(RTL6d1)@ The resulting @ProtocolMesssage@ must not exceed the @maxMessageSize@
-*** @(RTL6d2)@ Messages with differing @clientId@ values must not be bundled together
+*** @(RTL6d2)@ Messages with differing @clientId@ values must not be bundled together. (Note that this constraint only applies to what the client library can autonomously do as part of queuing messages, not to what the user can do by publishing an array of @Messages@. It exists because if any @Message@ in a @ProtocolMessage@ has an invalid @clientId@, the entire @ProtocolMessage@ is rejected. This is fine if the user has deliberately published the @Messages@ together – they requested atomicity – but not if the client library has bundled them without the user's knowledge)
```

Comment:
I think it should be explicit that messages can only be bundled if their clientIds are identical, or identically empty.

Reply:
Not sure I see the ambiguity (spec says "differing clientId values", `"foo"` and `null` are definitely not the same value). But shrug, guess it doesn't do any harm to be overexplicit